docs: update README to prepare for 1.0 release

walkawayy opened the pull request:
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [X] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description
Updating the README/docs for the upcoming TRX 1.0 release. I updated a lot of the showcase items and moved them to the general README and out of the TR1 section, tried to clarify TRX since the engine is now combined, and update the Q&A to reflect the current status. Lmk feedback, if I missed any common questions, what you think of the showcase, etc.

walkawayy committed:
docs: update README to prepare for 1.0 release

rr- commented on README.md:
```diff
@@ -1,5 +1,5 @@
 <div align="center">
-<h1>TRX – Tomb Raider I & II: Community Edition</h1>
```
I'd prefer that we don't try to present ourselves as the originals. Let's stick with the original header.

walkawayy replied:
Ya I get that. I was just thinking since we lost the community edition branding when we renamed to TRX, it didn't fit as much.

rr- commented on README.md:
```diff
@@ -90,28 +96,34 @@ Please refer to the [detailed documentation](docs/tr2/).
+    Regarding HD texture packs, that feature is not currently planned.
+
+    As for other visual updates, being able to introduce skyboxes and
+    reflections into TR1 shows that the sky is the limit. TRX is constantly
```
We lost the dad joke vibe along the way IMO

rr- reviewed (commented):
Apart from two nitpicks, LGTM

lahm86 commented on README.md:
```diff
@@ -1,5 +1,5 @@
-<h1>TRX – Tomb Raider I & II: Community Edition</h1>
+<h1>TRX – Tomb Raider I & II</h1>
 
 <a href="https://github.com/LostArtefacts/TRX/releases?q=tr1x+prerelease%3Afalse&expanded=true">
```
I think we'll just need the same link for both going forward i.e. no tag

https://github.com/LostArtefacts/TRX/releases?q=prerelease%3Afalse&expanded=true

walkawayy replied:
Oh true. I guess for the TR1 and TR2 sections, should we mention what download to look for? Idk if there's a cleaner way to do it. For non-technical people, there's a lot of platforms / links to sift through.

walkawayy replied:
Also @rr-, how do you want to handle the download images used such as `download_tr1x.svg`? We don't have a common `Download TRX` icon to use.

rr- replied:
We'll have to continue to provide two buttons as we ship separate bundles for TR1 and TR2. I can provide new SVGs soon, but ATM I don't have ideas other than just going icon-less, and with a bland "Download TR1 bundle", which isn't the most attractive option. Still at very least I'd avoid the previous branding "TR1X" "TR2X".

rr- replied:
Actually, the above is completely false – sorry.
The upcoming TRX 1.0 GitHub release, and the ones that will follow after, are going to be merged and contain assets for both TRs.
This means we have to adjust the links to the releases to cover this (`trx`), and offer a single button, as there's no way we can permalink to specific assets.

It actually simplifies things – with only one button, the necessity to visually distinguish between two buttons disappears, which excuses the bland look – I think going with just "Download TRX" is fine. The section about downloads can be then collapsed to

### Download

Download TRX [button]

Please see the following instructions on how to install the game:

- TR1 (link)
- TR2 (link)

(I'm all for making it prettier/more welcoming, but that's the essence of it)

You can see how this is going to look over here at my fork:
https://github.com/rr-/TRX/releases
(Scroll down to TRX-0.1)

walkawayy replied:
Pushed a bunch of updates to the `README.md` with a new download icon for TRX and renamed most branding from TR1/2X to say TRX for/targeting TR1/2 across many files. Also fixed a few README typos.

There are still places left across the project where TR1X and/or TR2X are still used. Some of that seems ok like the builder's migrating guide which talks about previous TR1X/TR2X versions. But here's a list of other stuff that uses the old names I wanted to make note of for posterity that I didn't touch and wanted to get opinions on:

1. Installers still use the TR1X/TR2X names thoughout the code, text, and the window name. What do we want to use in those?
<img width="768" height="608" alt="image" src="https://github.com/user-attachments/assets/04f5782d-d063-4e5d-8115-c244be15459c" />

2. Links such as `https://lostartefacts.dev/aux/tr1x/music.zip`, `https://lostartefacts.dev/aux/tr2x/trgm.zip` and a few others
3. Player settings saved in `cfg/TR1X.json5` and `cfg/TR2X.json5` with references to these in `docs/05-game_flow/4-USER_CONFIGURATION.md` and `docs/tr1/INSTALLING.md`
4. Console history `TR1X_console_history.json5` and `TR2X_console_history.json5`
5. Mac stuff such as `docs/SECRETS.md`, `data/tr2/mac/Info.plist`, and MacOS application ('darwin' section in `src/meson.build`). Macs might need to retain distinct names? Unsure about this one.
6. Old `CHANGELOG.md` entries (these seem ok since the project was called this back then though)
7. I wasn't sure if I should touch `TRX/src/trx/version.c` `const char *g_TRXVersion = "TR1X (non-Docker build)";`
8. Various code comments (such as `TRX/src/trx/config/priv.c`). I left these since they reference specific versions. I changed generic comments to use TRX.
9. `src/trx/game/level/reader_tr1.c` and `src/trx/game/level/reader_tr1.h` level loader stuff used elsewhere as well
10. `justfile.tr1` and `justfile.tr2`

rr- replied:
Apart from the installer the rest is not very important imo. Config files need to stay separate for now due to BC and differences in defaults. The installer should be updated though.

rr- replied:
And because of directory structure the installer needs to continue to look at the same directories as it does now, separate for tr1 and 2. Just the label and icon

lahm86 reviewed (commented):
Looks really nice! Just a comment on the download links.

walkawayy commented:
Pushed some changes based on the comments. Added the new TRX logo to the top which links to the website. Also added a blurb at the bottom about the website with the Lost Artefacts logo. Lmk what you think. I think the website should be mentioned somewhere. We just need to decide on what to do about the download links / images I believe.

rr- commented:
> Pushed some changes based on the comments. Added the new TRX logo to the top which links to the website. Also added a blurb at the bottom about the website with the Lost Artefacts logo. Lmk what you think. I think the website should be mentioned somewhere. We just need to decide on what to do about the download links / images I believe.

All looks good, thanks.

walkawayy committed:
further update docs to refer to TRX instead of TR1X or TR2X

rr- commented on README.md:
```diff
@@ -1,117 +1,136 @@
+</p>
+
+<a href="https://github.com/LostArtefacts/TRX/releases?q=prerelease%3Afalse&expanded=true">
+    <img src="data/download_trx.svg"/>
```
Missing asset , please lmk if you need me to edit the svg

walkawayy committed:
add download TRX common logo

walkawayy commented:
I noticed both the Windows and Mac manual install guides for TR1 and TR2 have the details section that needs updated to reflect the new shader files.

```
├── shaders
│   ├── 2d.glsl
│   ├── common.glsl
│   ├── fbo.glsl
│   └── meshes.glsl
```
